### DISCUSSION TITLE
Using JGroups on a crappy network for reliable comms

### DISCUSSION THREAD

**Discussion by snyderra** [belaban/JGroups#862] [Q&A] [answered]
Using JGroups on a network with ping times in the 5 seconds. What parameters can I tweak to make connection and member joining more stable.?

Have already adjusted GMS join timeout and FDSOCK2 connect timeout to 10 seconds

Base config is the default multicast (udp.xml)

Appreciate it!

**Comment by belaban** [comment, marked as answer]
Hi Rob

it depends... what transport are you using? UDP would probably be recommended over TCP, as the SYN-ACK-SYNACK connection establishment costs you dearly. Even worse when using TLS.

+1 on increasing the `GMS.join_timeout` and `FD_SOCK2.connect_timeout` timeouts. You might even consider removing `FD_SOCK2` altogether and only use `FD_ALL3`

If you use RPCs, make sure you set the call timeouts accordingly. Perhaps also use async RPCs in this case.

Is the latency fluctuating, or is it always 5s?
Also note that you could use `DELAY` to simulate this environment in your test env, if test env != prod env.

**Reply by snyderra** [reply]
Thanks! Removing FD_SOCK2 was the answer as the connections were consistently failing. Was using the UDP transport. 
For anyone else that needs this answer:
```java

Channel channel = JChannel(Global.DEFAULT_PROTOCOL_STACK);
channel.getProtocolStack().removeProtocol(FD_SOCK2.class);

GMS gms = channel.getProtocolStack().findProtocol(GMS.class);
if (gms!=null) {
    gms.setJoinTimeout(10000);
}
VERIFY_SUSPECT2 vs = channel.getProtocolStack().findProtocol(VERIFY_SUSPECT2.class);
if (vs!=null){
    vs.setValue("timeout",10000);
}
```

**Reply by belaban** [reply]
Great, good to know it works now!
